### PR TITLE
Added soap client

### PR DESCRIPTION
**Pull request by msjhall138**
Currently creating client service and posting to own service

### CODE REVIEW

File: src/main/java/com/kainos/example/ExternalApplication.java [hunk 1]
```diff
@@ -23,7 +25,7 @@ public void initialize(Bootstrap<ExternalConfiguration> bootstrap) {
 
     @Override
     public void run(ExternalConfiguration configuration,
-                    Environment environment) {
+                    Environment environment) throws MalformedURLException {
```
Comment (sdonaghey03) [inline src/main/java/com/kainos/example/ExternalApplication.java]: Catch this exception in the ThingClient

**Review by sdonaghey03 — CHANGES_REQUESTED** [review]

File: src/main/java/com/kainos/example/ExternalConfiguration.java [hunk 2]
```diff
@@ -15,4 +15,16 @@ public SoapServerConfig getSoapServer() {
...
+    private SoapClientConfig soapClient;
+
+    public SoapClientConfig getSoapClient() {
+        return soapClient;
+    }
+
+    public void setSoapClient(SoapClientConfig soapClient) {
+        this.soapClient = soapClient;
+    }
+
```
Comment (sdonaghey03) [inline src/main/java/com/kainos/example/ExternalConfiguration.java:29]: Remove blank line

File: src/main/java/com/kainos/example/ExternalApplication.java [hunk 3]
```diff
@@ -32,5 +33,11 @@ public void run(ExternalConfiguration configuration,
         environment.healthChecks().register("alive", applicationHealthCheck);
 
         Endpoint.publish(configuration.getSoapServer().getBaseUrl(), new ThingService());
+
+        ThingClient thingClient = ThingClient.getInstance(configuration);
+        IThingService ts = thingClient.getClientService();
```
Comment (simonwatsonkainos) [inline src/main/java/com/kainos/example/ExternalApplication.java:38]: Can we have a better variable name?

**Review by sdonaghey03 — CHANGES_REQUESTED** [review]

File: src/main/java/com/kainos/example/jaxws/client/ThingClient.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public static ThingClient getInstance (ExternalConfiguration configuration) {
+        if (instance == null) {
+            instance = new ThingClient();
+            instance.initialise(configuration);
+        }
+        return instance;
+    }
+
+    public IThingService getClientService () {
```
Comment (sdonaghey03) [inline src/main/java/com/kainos/example/jaxws/client/ThingClient.java:39]: This should probably be getThingService

**Pull request closed without merging**